### PR TITLE
Workflow history log / progress views

### PR DESCRIPTION
**Pull request by kaedroho**

### CODE REVIEW

File: wagtail/admin/templates/wagtailadmin/pages/workflow_history/detail.html [hunk 1]
```diff
@@ -1,21 +1,20 @@
 {% extends "wagtailadmin/base.html" %}
 {% load i18n wagtailadmin_tags %}
 
-{% block titletag %}{% blocktrans with title=page.get_admin_display_title %}Workflow progress {{ title }}{% endblocktrans %}{% endblock %}
+{% block titletag %}{% blocktrans %}Workflow progress{% endblocktrans %}{% endblock %}
```
Comment (zerolab) [inline wagtail/admin/templates/wagtailadmin/pages/workflow_history/detail.html]: this would be better as `{% trans "Workflow progress" %}`

Comment (kaedroho) [reply wagtail/admin/templates/wagtailadmin/pages/workflow_history/detail.html]: Oh yes, thanks!

File: wagtail/admin/templates/wagtailadmin/shared/workflow_status.html [hunk 2]
```diff
@@ -24,11 +24,13 @@ <h2>
         {% trans 'Approved task' as approved_title %}
         {% trans 'Incomplete task' as incomplete_title %}
         {% for task in workflow_tasks %}
-            {% if task.status == 'approved' %}
-                {% wagtail_icon name="success" title=approved_title %}
-            {% else %}
-                {% wagtail_icon name="radio-empty" title=incomplete_title %}
-            {% endif %}
+            <span data-wagtail-tooltip="{{ task.name }}: {{ task.status_display }}">
```
Comment (jacobtoppm) [inline wagtail/admin/templates/wagtailadmin/shared/workflow_status.html]: I think this wouldn't be screen reader accessible - possibly we could follow one of the approaches here, under toggletips? https://inclusive-components.design/tooltips-toggletips/

Comment (jacobtoppm) [reply wagtail/admin/templates/wagtailadmin/shared/workflow_status.html]: At the very least make this info accessible on the icon instead, now we have it in the context

File: wagtail/admin/templates/wagtailadmin/pages/workflow_history/list.html [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                        {% for task in workflow_state.all_tasks_with_status %}
+                            {% if task.status == 'approved' %}
+                                {% wagtail_icon name="success" title=approved_title %}
+                            {% else %}
+                                {% wagtail_icon name="radio-empty" title=incomplete_title %}
+                            {% endif %}
+                        {% endfor %}
+                    </td>
+                    <td>{{ workflow_state.requested_by }}</td>
+                    <td>{{ workflow_state.created_at }}</td>
```
[the remark this answers is not in the document]
Comment (kaedroho) [reply wagtail/admin/templates/wagtailadmin/pages/workflow_history/list.html]: Yep, might as well. Also spotted the field before needs to get the user's name.

There are a bunch of places where we could do this on the workflow progress view, but we probably should decide that in UX review (date formatting seems to be quite inconsistent throughout Wagtail at the moment)

File: wagtail/admin/templates/wagtailadmin/pages/workflow_history/detail.html [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+                    document.querySelectorAll('.tab-buttons button').forEach(tab => tab.classList.remove('active'));
+
+                    button.classList.add('active');
+                });
+            });
+        });
+    </script>
+
+    <style>
+        .tab-buttons button {
```
[the remark this answers is not in the document]
Comment (jacobtoppm) [reply wagtail/admin/templates/wagtailadmin/pages/workflow_history/detail.html]: Great, thanks for doing this!

**Pull request merged**